awx/api/serializers.py

```diff
@@ -4777,12 +4778,9 @@ def check_organization_permission(self, attrs, request):
                         raise ValidationError(_(f"Organization {requested_org.id} not found or you don't have permissions to access it"))
 
     def check_unified_job_permission(self, request, requested_ujts):
-        allowed_ujts = set()
-        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'execute_role').all()]
-        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'admin_role').all()]
-        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'update_role').all()]
-        accessible_inventories_qs = Inventory.accessible_pk_qs(request.user, 'update_role')
-        [allowed_ujts.add(tup[0]) for tup in InventorySource.objects.filter(inventory__in=accessible_inventories_qs).values_list('id')]
+        allowed_jts = set(JobTemplate.accessible_pk_qs(request.user, 'execute_role').values_list('id', flat=True))
+        allowed_inv_sources = set(InventorySource.objects.filter(inventory__in=Inventory.accessible_pk_qs(request.user, 'update_role')).values_list('id'))
+        allowed_ujts = allowed_jts | allowed_inv_sources
```

Comment:
I got here by a test case of launching a _system job template_. It did not work, and fell into this error. The help text now reflects that you can only launch JTs and update inventory sources.

Workflows do not support system jobs. I'm hesitant to support workflows themselves here.

You could combine this into a single queryset, but I'd rather not, depending on the feature intent.

Reply:
what about project updates?

Reply:
I did not know you were supporting these when I did this. I updated the diff to work with those, but now it's an inconsistent approach compared to the other checks.

If you don't like this approach, I would suggest that you move `get_objectified_jobs` earlier up in the validate method. Then you can sort the UJTs by type, and then do your check on each type one-by-one.